client/components/search/index.js

```diff
@@ -71,55 +78,111 @@ class Search extends Component {
+							label={ item.label }
+							remove={ this.removeResult }
+							removeLabel={ __( 'Remove product', 'wc-admin' ) }
+							screenReaderLabel={ screenReaderLabel }
+						/>
+					);
+				} ) }
+			</div>
+		) : null;
+	}
+
+	onFocus() {
+		this.setState( { isActive: true } );
+	}
+
+	onBlur() {
+		this.setState( { isActive: false } );
+	}
+
 	render() {
 		const autocompleter = this.getAutocompleter();
-		const { placeholder, selected } = this.props;
-		const { value = '' } = this.state;
+		const { placeholder, inlineTags, selected, instanceId } = this.props;
+		const { value = '', isActive } = this.state;
 		const aria = {
 			'aria-labelledby': this.props[ 'aria-labelledby' ],
 			'aria-label': this.props[ 'aria-label' ],
 		};
 		return (
 			<div className="woocommerce-search">
+				<label htmlFor={ `search-input-${ instanceId }` } className="screen-reader-text">
+					{ placeholder || __( 'Add item', 'wc-admin' ) }
+				</label>
 				<Gridicon className="woocommerce-search__icon" icon="search" size={ 18 } />
 				<Autocomplete completer={ autocompleter } onSelect={ this.selectResult }>
-					{ ( { listBoxId, activeId, onChange } ) => (
-						<input
-							type="search"
-							value={ value }
-							placeholder={ placeholder }
-							className="woocommerce-search__input"
-							onChange={ this.updateSearch( onChange ) }
-							aria-owns={ listBoxId }
-							aria-activedescendant={ activeId }
-							{ ...aria }
-						/>
-					) }
-				</Autocomplete>
-				{ selected.length ? (
-					<div className="woocommerce-search__token-list">
-						{ selected.map( ( item, i ) => {
-							const screenReaderLabel = sprintf(
-								__( '%1$s (%2$s of %3$s)', 'wc-admin' ),
-								item.label,
-								i,
-								selected.length
-							);
-							return (
-								<Tag
-									key={ item.id }
-									id={ item.id }
-									label={ item.label }
-									remove={ this.removeResult }
-									removeLabel={ __( 'Remove product', 'wc-admin' ) }
-									screenReaderLabel={ screenReaderLabel }
+					{ ( { listBoxId, activeId, onChange } ) =>
+						// Disable reason for no-static-element-interactions: There is no appropriate role
+						// which describes the input container intended accessible usability.
+						// Disable reason for click-events-have-key-events: The div below visually
+						// simulates an input field. Its child input is the actual input and
+						// responds accordingly to all keyboard events, but click events need to be
+						// passed onto the child input.
+						/* eslint-disable jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events */
+						inlineTags ? (
+							<div
+								className={ classnames( 'woocommerce-search__inline-container', {
+									'is-active': isActive,
+								} ) }
+								onClick={ () => {
+									this.input.current.focus();
+								} }
+							>
+								{ this.renderTags() }
+								<input
+									ref={ this.input }
+									type="text"
```

Comment:
why is this type=text here and type=search below?

Reply:
"search" tells the browser to render the clear button on the right:

![screen shot 2018-10-12 at 9 59 52 am](https://user-images.githubusercontent.com/1922453/46833804-22933b00-ce06-11e8-972b-bd5080e05b70.png)

I don't think it makes sense in this context to have a clear button. Would it clear the text? Or all tags?

Reply:
Ah gotcha, no I think **not** having the browser clear button makes sense for inline tags as it would likely cause confusion.